aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -452,16 +414,23 @@ public void onError(Exception exception) {
     }
 
     @Override
-    public void forgotPassword(
+    public void resetPassword(
             @NonNull String username,
-            @NonNull Consumer<AuthCodeDeliveryDetails> onSuccess,
+            @NonNull Consumer<AuthResetPasswordResult> onSuccess,
             @NonNull Consumer<AuthException> onException
     ) {
-        AWSMobileClient.getInstance().forgotPassword(username, new Callback<ForgotPasswordResult>() {
+        awsMobileClient.forgotPassword(username, new Callback<ForgotPasswordResult>() {
             @Override
             public void onResult(ForgotPasswordResult result) {
                 if (result.getState().equals(ForgotPasswordState.CONFIRMATION_CODE)) {
-                    onSuccess.accept(convertCodeDeliveryDetails(result.getParameters()));
+                    onSuccess.accept(new AuthResetPasswordResult(
+                            false,
```

Comment:
The thing about this is that we do want the value being stored and accessed by the customer to be a boolean vs. an enum and a customer would never need to construct this result object so in this case, I think the extra complexity this would introduce isn't justified.

Reply:
Per the design meeting, the value the customer is interacting with should be a boolean (in this case isPasswordReset) so given that plugin developers are constructing this as a result object, I think it's more straightforward for them to construct it with the raw values they intend the customer to receive rather than having to check the implementation of isPasswordReset() to see that while it's constructed with an ENUM, it ends up being converted to a boolean result.